### PR TITLE
Drop pg 10 from Nix

### PR DESCRIPTION
**Pull request by steve-chavez**
Using the flake template introduced on https://github.com/tcdi/pgx/pull/148. I get the following error:

```
$ nix develop
error: postgresql_10 has been removed from nixpkgs, as this version went EOL on 2022-11-10
```

With this change the error goes away.

```
$ nix develop
$ cargo pgx -V
cargo-pgx 0.5.6
```

I think this goes in line with https://github.com/tcdi/pgx/pull/830.

(Also removed some Nix deps from `cargo-pgx` that look unused)

### CODE REVIEW

**Review by workingjubilee — APPROVED** [review]
Thank you!

File: cargo-pgx/default.nix [hunk 1]
```diff
@@ -1,14 +1,8 @@
...
 , naersk
 , hostPlatform
-, fetchFromGitHub
-, postgresql_10
-, postgresql_11
-, postgresql_12
-, postgresql_13
 , pkg-config
 , openssl
-, rustfmt
```
Comment (workingjubilee) [inline cargo-pgx/default.nix:11]: Actually, wait. I know they look unused but can you reduce this to the most conservative change?

File: cargo-pgx/default.nix [hunk 2]
```diff
@@ -1,14 +1,8 @@
 { lib
 , naersk
 , hostPlatform
-, fetchFromGitHub
```
Comment (workingjubilee) [inline cargo-pgx/default.nix:4]: This one is actually unused, yes, it can go.

File: cargo-pgx/default.nix (hunk 1 above)
Comment (steve-chavez) [reply cargo-pgx/default.nix:11]: Sure, should I just remove 0f0a1618e2dca0f4b9c3f4057da841c6f4df3358 from this PR or add another commit that only removes `fetchFromGitHub`?

Comment (workingjubilee) [reply cargo-pgx/default.nix:11]: Pushing another commit is fine.

Comment (workingjubilee) [reply cargo-pgx/default.nix]: Apologies, I am on a timer.

**Pull request merged**